A43-sdk-authorization.md

```diff
@@ -0,0 +1,424 @@
+					"type": "array",
+					"items": {
+					  "type": "string"
+					}
+				  }
+				},
+				"required": ["key", "values"],
+		      }
+			}
+		  }
+		}
+	  },
+	  "required": ["name"]
+	}
+  },
+  "description": "AuthorizationPolicy defines which principals are permitted to access which resource. Resources are RPC methods"
+    "scoped by services.",
+  "type": "object",
+  "properties": {
+	"name": {
+	  "description": "The name of an authorization policy. It is mainly for monitoring and error message generation.",
+	  "type": "string",
+	},
+	"deny_rules": {
+      "description": "List of deny rules to match. If a request matches any of the deny rules, then it will be denied. If none of"
+		"the deny rules matches or there are no deny rules, the allow rules will be evaluated.",
+	  "type": "array",
+      "items": {
+	    "$ref": "#/definitions/rule",
+	  },
+	},
+	"allow_rules": {
+      "description": "List of allow rules to match. The allow rules will only be evaluated after the deny rules. If a request"
+	    "matches any of the allow rules, then it will allowed. If none of the allow rules matches, it will be denied.",
+	  "type": "array",
+	  "items": {
+		"$ref": "#/definitions/rule",
+	  },
+	}
+  },
+  "required": ["name", "allow_rules"]
+}
+```
+
+#### Details
+
+gRPC SDK authorization policy has a list of allow rules and a list of deny rules. 
+The following sequence is followed to make an authorization decision -
+1. Check for a match in deny rules, if a match is found, the request will be denied. 
+   If no match is found in deny rules, or there are no deny rules execute next step.
+2. Check for a match in allow rules, if a match is found, the request will be allowed. 
+   Note that allow rules is a required field here, if not present, policy is invalid.
+3. If no match is found in allow rules, we deny the request.
+
+Each rule has the following semantics -
+1. Each rule has a name, a source and a request to match against. For a rule to match,
+   both source and request must match.
+   - If both source and request are empty, and the policy is an allow list, we will
+     accept all RPC requests. On the other hand, if the policy were a denylist, we
+     would deny all RPC requests.
+   - If only source is empty, we evaluate against the request fields and apply that to
+     any user (See example below). Similarly, if only request is empty, we evaluate
+     against the source fields and apply that to any action.
+   - This also applies to sub fields, like if the principals in source is empty, the
+     behavior would be similar to an empty source.
+2. Each source could contain a list of principals. The principals are ORed together,
+   i.e. it matches if one of them matches.
+   Sequence of steps to evaluate each principal from config -
+   1. If TLS is not used, matching fails.
+   2. A match is found if principal matches any one of URI SANs or DNS SANs or Subject
```

Comment:
This is different behavior than RBAC. In RBAC, URI is used if there is such a SAN, then DNS, and Subject is only checked if there were no URI or DNS SANs.

Reply:
Sorry for the confusing wording. I was trying to say the same thing. Updated text. PTAL.